client/client.go

```diff
@@ -235,83 +242,88 @@ func (c *CfClient) authenticate(ctx context.Context, target evaluation.Target) {
 	payload := strings.Split(c.token, ".")[payloadIndex]
 	payloadData, err := jwt.DecodeSegment(payload)
 	if err != nil {
-		c.config.Logger.Error(err)
-		return
+		return err
 	}
 
 	var claims map[string]interface{}
 	if err = json.Unmarshal(payloadData, &claims); err != nil {
-		c.config.Logger.Error(err)
-		return
+		return err
 	}
 
 	var ok bool
 	c.environmentID, ok = claims["environment"].(string)
 	if !ok {
-		c.config.Logger.Error(errors.New("environment uuid not present"))
-		return
+		return fmt.Errorf("environment uuid not present")
 	}
 
 	c.clusterIdentifier, ok = claims["clusterIdentifier"].(string)
 	if !ok {
-		c.config.Logger.Error(errors.New("cluster identifier not present"))
 		c.clusterIdentifier = "1"
+		return fmt.Errorf("cluster identifier not present")
 	}
 
 	// network layer setup
 	bearerTokenProvider, bearerTokenProviderErr := securityprovider.NewSecurityProviderBearerToken(c.token)
 	if bearerTokenProviderErr != nil {
-		c.config.Logger.Error(bearerTokenProviderErr)
-		return
+		return bearerTokenProviderErr
 	}
 	restClient, err := rest.NewClientWithResponses(c.config.url,
 		rest.WithRequestEditorFn(bearerTokenProvider.Intercept),
 		rest.WithRequestEditorFn(c.InterceptAddCluster),
 		rest.WithHTTPClient(c.config.httpClient),
 	)
 	if err != nil {
-		c.config.Logger.Error(err)
-		return
+		return err
 	}
 
 	merticsClient, err := metricsclient.NewClientWithResponses(c.config.eventsURL,
 		metricsclient.WithRequestEditorFn(bearerTokenProvider.Intercept),
 		metricsclient.WithHTTPClient(http.DefaultClient),
 	)
 	if err != nil {
-		c.config.Logger.Error(err)
-		return
+		return err
 	}
 
 	c.api = restClient
 	c.metricsapi = merticsClient
+	c.config.Logger.Info("Authentication complete")
 	close(c.authenticated)
+	return nil
 }
 
 func (c *CfClient) makeTicker(interval uint) *time.Ticker {
 	return time.NewTicker(time.Minute * time.Duration(interval))
 }
 
 func (c *CfClient) pullCronJob(ctx context.Context) {
+	poll := func() {
+		c.mux.RLock()
+		if !c.streamConnected {
+			ok := c.retrieve(ctx)
```

Comment:
if we fail to retrieve what do we do?

Reply:
We just log the error out in that retrieve function then the next poll interval we try again

Reply:
sweet